Jenkinsfile

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env groovy
+
+// These are the platforms we are building against.
+def platforms = ["linux", "mac", "windows"]
+
+// This generates the launchConfig for a particular build.
+def generateConfig(deploymentName, templateUrl) {
+    return """
+---
+launch_config_version: 1
+deployment_name: ${deploymentName}
+template_url: ${templateUrl}
+provider: aws
+aws_region: us-west-1
+template_parameters:
+    KeyName: default
+    AdminLocation: 0.0.0.0/0
+    PublicSlaveInstanceCount: 1
+    SlaveInstanceCount: 1
+"""
+}
+
+
+// This class abstracts away the functions required to create a test cluster
+// for each of our platforms.
+class TestCluster implements Serializable {
+    WorkflowScript script
+    String platform
+    int createAttempts
+
+    TestCluster(WorkflowScript script, String platform) {
+        this.script = script
+        this.platform = platform
+        this.createAttempts = 0
+    }
+
+    def _create() {
```

Comment:
I wanted to have simple wrappers around just the sh commands instead of embedding them in more complicated logic for the "real" variants of those commands. The exception is delete, which didn't need a wrapper, so didn't have an '_' version. The '_' itself was meant to imply that they shouldn't use used outside the class. I could probably use the private keyword for that though.